pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/PipInstallTask.groovy

```diff
@@ -134,81 +141,105 @@ class PipInstallTask extends DefaultTask implements FailureReasonProvider {
-
-                def startTime = new Date()
-                def stream = new ByteArrayOutputStream()
-                ExecResult installResult = project.exec { ExecSpec execSpec ->
-                    execSpec.environment mergedEnv
-                    execSpec.commandLine(commandLine)
-                    execSpec.standardOutput = stream
-                    execSpec.errorOutput = stream
-                    execSpec.ignoreExitValue = true
-                }
-                def endTime = new Date()
-                def duration = TimeCategory.minus(endTime, startTime)
-
-                def message = stream.toString().trim()
-                if (installResult.exitValue != 0) {
-                    /*
-                     * TODO: maintain a list of packages that failed to install, and report a failure
-                     * report at the end. We can leverage our domain expertise here to provide very
-                     * meaningful errors. E.g., we see lxml failed to install, do you have libxml2
-                     * installed? E.g., we see pyOpenSSL>0.15 failed to install, do you have libffi
-                     * installed?
-                     */
-                    logger.lifecycle(message)
-                    lastInstallMessage = message
-
-                    throw new PipInstallException(
-                        "Failed to install ${shortHand}. Please see above output for reason, or re-run your build using ``gradle -i build`` for additional logging.")
-                } else {
-                    if (extension.consoleOutput == ConsoleOutput.RAW) {
-                        logger.lifecycle(message)
-                    } else {
-                        String prefix = String.format("Install (%d:%02d.%03d s)", duration.minutes, duration.seconds % 60, duration.millis % 1000)
-                        logger.lifecycle(PythonHelpers.createPrettyLine(prefix, "[FINISHED]"))
-                    }
-                }
+                String shortHand = packageInfo.version ? "${ packageInfo.name }-${ packageInfo.version }" : packageInfo.name
+
+                def timer = taskTimer.start(shortHand)
+                logger.info("Installing {}", shortHand)
+                progressLogger.progress("Installing $shortHand (${ ++counter } of ${ installableFiles.size() })")
+                doInstall(shortHand, packageInfo, sitePackages, pyVersion, extension, installable)
+                timer.stop()
+            }
+        }
+
+        progressLogger.completed()
+
+        new File(project.buildDir, getName() + "-task-runtime-report.txt").text = taskTimer.buildReport()
+    }
+
+    @SuppressWarnings("ParameterCount")
+    private void doInstall(String shortHand, PackageInfo packageInfo, Path sitePackages,
+                           String pyVersion, PythonExtension extension, File installable) {
+        if (packageExcludeFilter.isSatisfiedBy(packageInfo)) {
+            return
```

Comment:
Loggin of excluded packages really helps with reasoning about the build failure for conditional dependencies between Python 2 and 3. Can we put the logging back please?

Reply:
Done.